reference/tools.rst

```diff
@@ -591,7 +591,7 @@ tools.patch()
     def patch(base_path=None, patch_file=None, patch_string=None, strip=0, output=None, fuzz=False)
 
 Applies a patch from a file or from a string into the given path. The patch should be in diff (unified diff) format. To be used mainly in
-the ``source()`` method.
+the ``build()`` method.
```

Comment:
I'd try to add a reason here, very short, but something:

_"Use this tool in the ``source()`` method only for those patches that don't depend on settings, otherwise move usage to ``build()`` method"_

Reply:
I would say, only when is not copied to build_folder, `no_copy_sorce=True`. Otherwise, use build_folder.

Reply:
Still no, even if you use ``no_copy_source=True``, you don't want to try to apply patches that depends on settings or options.

Reply:
For recipe development, it's best applying to `build()` only, even when it doesn't depends on settings.
- When you need to check the authors source code to review, the source folder is the clear, you can check before any patch there.
- We need to download the sources only once, if for some reason the patch is broken/wrong, you will need to download all sources again until fixing it, and believe, it happens.
- build folder is re-created for every new build, applying patch there is not problem  because it's "temporary"
- IHMO source folder should be untouched. we can use for review as the single source of truth

The only case that I can't use the build folder, is for `no_copy_source=True`. Because the package folder needs to collect from source folder only. Header-only and Boost are good examples.

Reply:
Ok, then prefer the `build()` over the `source()`: _"Use it preferably in the `build()` method"_ 👍 

The `source()` method won't have `settings` and `options` available ([docs about v2](https://docs.conan.io/en/latest/reference/conan_v2_mode.html#changes-in-recipes)), so those recipes with `no_copy_sorce=True` won't be able to use it in order to apply conditionally some patches.

Reply:
Done, I just added your new statement. It sounds good.